src/tests/JIT/Regression/JitBlue/Runtime_105817/Runtime_105817.cs

```diff
@@ -0,0 +1,60 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Generated by Fuzzlyn v2.2 on 2024-08-01 14:37:47
+// Run on Arm64 MacOS
+// Seed: 14773448547728333023-vectort,vector64,vector128,armadvsimd,armadvsimdarm64,armaes,armarmbase,armarmbasearm64,armcrc32,armcrc32arm64,armdp,armrdm,armrdmarm64,armsha1,armsha256
+// Reduced from 270.2 KiB to 0.4 KiB in 00:01:48
+// Debug: Outputs 0
+// Release: Outputs 1
+using System;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using Xunit;
+
+public class Runtime_105817
+{
+    [Fact]
+    public static void TestOverShiftRightLogical()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr6 = Vector128.Create<short>(1);
+            var vr7 = AdvSimd.ShiftRightLogical(vr6, 16);
+            Assert.Equal(vr7, Vector128<short>.Zero);
+        }
+    }
+
+    [Fact]
+    public static void TestOverShiftRightLogicalScalar()
+    {
+        if (AdvSimd.IsSupported)
+        {
+            var vr6 = Vector64.Create<long>(1);
+            var vr7 = AdvSimd.ShiftRightLogicalScalar(vr6, 64);
```

Comment:
Can we also check an out of range value like `65`? With the other change it should now be executing the `ShiftLogical` path rather than throwing, so we want to ensure its computing the right result between the two scenarios as well.

Reply:
Sure thing. #105777 only added the fallback implementation for `ShiftRightLogical`, so `ShiftRightLogicalScalar`, `ShiftRightArithmetic`, and `ShiftRightArithmeticScalar` still throw in Debug. Are you ok with me adding their fallback implementations in this PR, or would you prefer that work in a separate one?

Reply:
We weren't using the fallback APIs for non-const immediates either, so I added the `HW_Flag_NoJmpTableIMM` flag to the `ShiftRight*` APIs so we always use the fallback instead of a jump table.